### PR TITLE
fix: Render <important> and <note> XML doc tags in generated API reference

### PR DESCRIPTION
**Pull request by AlexDaines**
## Description

Add handling for `<important>` and `<note>` XML doc tags in the doc generator's HTML transformation pipeline. These tags appear across 2,583+ occurrences in the SDK's XML documentation corpus but were previously passed through as unrecognized elements. They now render as styled `<div class="noteblock">` blocks with an appropriate header (`Important:` or `Note:`).

## Motivation and Context

Service teams use `<important>` and `<note>` tags in their API model documentation to highlight critical information. The doc generator was silently passing these through as raw `<important>`/`<note>` HTML elements, which browsers ignore as unknown tags — the content rendered as unstyled inline text with no visual distinction.

### Dry-run - Powershell5
- **Dry-run (build) ID:** DRY_RUN-74b028b4-df80-443f-8f11-5c997bcf9c45
- **Status:**
  - [ ] Pending
  - [x] Completed successfully
  - [ ] Failed
- **Failed bypass reason:**

### Dry-run - Dotnet4
- **Dry-run (build) ID:** DRY_RUN-7db66e91-4d87-448b-bfa5-382f29243673
- **Status:**
  - [ ] Pending
  - [x] Completed successfully
  - [ ] Failed
- **Failed bypass reason:**

5 unit tests added in `NDocUtilitiesHtmlTests.cs`, all using exact output assertions (`Assert.Equal`):

- `ImportantTag_RendersAsNoteblock` — standard `<important>` with inner `<para>`
- `NoteTag_RendersAsNoteblock` — standard `<note>` with inner `<para>`
- `ParaTag_StillRendersAsParagraph` — regression test for existing element mapping
- `SelfClosingImportantTag_ProducesEmptyNoteblock` — self-closing `<important />` edge case
- `ImportantTagWithInnerMarkup_PreservesNestedElements` — `<c>` tags nested inside callout, matching real corpus patterns

All 10 tests in `SDKDocGenerator.UnitTests` pass.

## Screenshots (if appropriate)

image

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [x] My code follows the code style of this project
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly
- [x] I have read the **README** document
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed

## License
- [x] I confirm that this pull request can be released under the Apache 2 license

### CODE REVIEW

**Comment by dscpinheiro** [comment]
Screenshots on this PR would be helpful to understand what the pages will look like (you can get the artifacts from the doc updater step).

**Comment by AlexDaines** [comment]
> Screenshots on this PR would be helpful to understand what the pages will look like (you can get the artifacts from the doc updater step).

Good idea. Added an example screenshot

**Comment by muhammad-othman** [comment]
I thought we will use `<strong>` tag or something to convey the importance of these paragraphs, adding `Important:` seems overly verbose.

**Comment by peterrsongg** [comment]
I do kind of agree with Muhammad's general comment that the <important> tag should wrap the text in bold using <strong> or something like that. Don't feel too strongly about keeping or removing "Important" but i think the text should be bolded at least

**Comment by AlexDaines** [comment]
> I thought we will use <strong> tag or something to convey the importance of these paragraphs, adding Important: seems overly verbose.

> I do kind of agree with Muhammad's general comment that the tag should wrap the text in bold using ** or something like that. Don't feel too strongly about keeping or removing "Important" but i think the text should be bolded at least**

I agree with this assessment. I went ahead and revised to use css changes to emphasize <important> and <note> sections and updated screenshots with the new implementation.

File: docgenerator/SDKDocGeneratorLib/NDocUtilities.cs [hunk 1]
```diff
@@ -604,18 +612,26 @@ private static string DocBlobToHTML(XElement rootNode, AbstractTypeProvider type
...
+                                    elementName = originalLocalName;
 
                                 // some elements can't be empty, use this variable for that
                                 string emptyElementContents = null;
 
                                 // start element
                                 writer.WriteStartElement(elementName);
 
+                                // Add CSS class if the original element has a class mapping
+                                string cssClass;
```
Comment (muhammad-othman) [inline docgenerator/SDKDocGeneratorLib/NDocUtilities.cs:629]: nit: not need to define the class beforehand as it isn't used anywhere outside the if condition, you can do this instead

```
                                if (NdocToHtmlClassMapping.TryGetValue(originalLocalName, out var cssClass))
                                {
                                    writer.WriteAttributeString("class", cssClass);
                                }
```

File: docgenerator/SDKDocGeneratorLib/NDocUtilities.cs [hunk 2]
```diff
@@ -41,7 +41,15 @@ public static class NDocUtilities
             { "summary", "p" },
             { "para", "p" },
             { "see", "a" },
-            { "paramref", "code" }
+            { "paramref", "code" },
+            { "important", "div" },
```
Comment (muhammad-othman) [inline docgenerator/SDKDocGeneratorLib/NDocUtilities.cs:45]: I would prefer if we use another element for `important` other than `div` for better accessibility.

File: docgenerator/SDKDocGeneratorLib/NDocUtilities.cs [hunk 3]
```diff
@@ -41,7 +41,15 @@ public static class NDocUtilities
...
+            { "note", "div" }
```
Comment (muhammad-othman) [inline docgenerator/SDKDocGeneratorLib/NDocUtilities.cs:46]: maybe we can even use `small` for `note` and if we did so we wont need `NdocToHtmlClassMapping` as `important`&`note` can be identified with html tags and no need for css classes.

**Review by muhammad-othman — APPROVED** [review]
Added few other comments but they are suggestions and not blocking so I'm approving the PR anyway.
